### PR TITLE
Encode all ATI parameters

### LINKED ISSUE #7757: Marketing ATI tracking not working for some URLs
**Describe the bug**
Marketing have reported that some URLs are not reporting to ATI correctly.

**To Reproduce**

1. Go to `https://www.bbc.com/tamil/india-53826135?xtor=CS3-33-%5Bwstamil%7EC%7EA38B38C42D38E38F39G38wsmktgProudToBeTamil%7ETamil_8MS_ProudToBeTamil_Facebook_Traffic_India_E38F39G38%5D-%5BFacebook%5D-%5B23845932568300152%5D-%5B23845937811820152%5D`
2. Open the Network Tab and filter for `a1.`
3. See the ATI call and examine it's parameters
4. Note absence of an `xto` value

**Expected behaviour**
We should see `&xto=CS3-33-[wstamil~C~A38B38C42D38E38F39G38wsmktgProudToBeTamil~Tamil_8MS_ProudToBeTamil_Facebook_Traffic_India_E38F39G38]-[Facebook]-[23845932568300152]-[23845937811820152]` in the ATI call

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by andrewscfc**
Resolves https://github.com/bbc/simorgh/issues/7757

**Overall change:**
The bug was caused by an unescaped `?` the x9 value corrupting the query string and causing ati to not receive any values after the x9 parameter:
x9=[%E0%AE%95%E0%AE%B0%E0%AE%BF%E0%AE%95%E0%AE%BE%E0%AE%B2%E0%AE%A9%E0%AF%8D+%E0%AE%95%E0%AE%9F%E0%AF%8D%E0%AE%9F%E0%AE%BF%E0%AE%AF+%E0%AE%95%E0%AE%B2%E0%AF%8D%E0%AE%B2%E0%AE%A3%E0%AF%88:+%E0%AE%A4%E0%AE%AE%E0%AE%BF%E0%AE%B4%E0%AE%B0%E0%AF%8D%E0%AE%95%E0%AE%B3%E0%AE%BF%E0%AE%A9%E0%AF%8D+%E0%AE%A8%E0%AF%80%E0%AE%B0%E0%AF%8D+%E0%AE%AE%E0%AF%87%E0%AE%B2%E0%AE%BE%E0%AE%A3%E0%AF%8D%E0%AE%AE%E0%AF%88+%E0%AE%95%E0%AF%81%E0%AE%B1%E0%AE%BF%E0%AE%A4%E0%AF%8D%E0%AE%A4%E0%AF%81+%E0%AE%A8%E0%AF%80%E0%AE%99%E0%AF%8D%E0%AE%95%E0%AE%B3%E0%AF%8D+%E0%AE%85%E0%AE%B1%E0%AE%BF%E0%AE%B5%E0%AF%80%E0%AE%B0%E0%AF%8D%E0%AE%95%E0%AE%B3%E0%AE%BE `?` +#%E0%AE%A4%E0%AE%AE%E0%AE%BF%E0%AE%B4%E0%AE%B0%E0%AF%8D_%E0%AE%AA%E0%AF%86%E0%AE%B0%E0%AF%81%E0%AE%AE%E0%AF%88+-+BBC+News+%E0%AE%A4%E0%AE%AE%E0%AE%BF%E0%AE%B4%E0%AF%8D]

This PR moves to encode all ati parameters before a hit is dispatched, I don't believe there is any reason we should not encode all our parameters, an alternative to this (arguably) more comprehensive PR is to just encode the x9 parameter and leave other parameters unchanged. I believe this PR will make things more consistent and decrease the likelihood of other encoding related issues being introduced in the future.

The below fully decoded hit is received in the ATI tag inspector shown in a screenshot below (x5 & x6 are intentionally double encoded see https://github.com/bbc/simorgh/pull/6593): 

<details>
  <summary>full hit</summary>
https://logws1363.ati-host.net/?s=598343&idclient=f90361a3-421d-4da3-a110-131eab73e794&s2=87&p=india%3A%3Atamil.india.story.53826135.page&r=1920x1080x24x24&re=150x898&hl=17x13x2&lng=en-GB&x1=[urn%3Abbc%3Acps%3Acebacc47-6b26-413e-9ebf-196bf4fb0b67]&x2=[responsive]&x3=[news-tamil]&x4=[ta]&x5=[http%253A%252F%252Flocalhost%253A7080%252Ftamil%252Findia-53826135%253Fxtor%253DCS3-33-%255Bwstamil~C~A38B38C42D38E38F39G38wsmktgProudToBeTamil~Tamil_8MS_ProudToBeTamil_Facebook_Traffic_India_E38F39G38%255D-%255BFacebook%255D-%255B23845932568300152%255D-%255B23845937811820152%255D]&x7=[article]&x8=[simorgh]&x9=[%E0%AE%95%E0%AE%B0%E0%AE%BF%E0%AE%95%E0%AE%BE%E0%AE%B2%E0%AE%A9%E0%AF%8D%2B%E0%AE%95%E0%AE%9F%E0%AF%8D%E0%AE%9F%E0%AE%BF%E0%AE%AF%2B%E0%AE%95%E0%AE%B2%E0%AF%8D%E0%AE%B2%E0%AE%A3%E0%AF%88%3A%2B%E0%AE%A4%E0%AE%AE%E0%AE%BF%E0%AE%B4%E0%AE%B0%E0%AF%8D%E0%AE%95%E0%AE%B3%E0%AE%BF%E0%AE%A9%E0%AF%8D%2B%E0%AE%A8%E0%AF%80%E0%AE%B0%E0%AF%8D%2B%E0%AE%AE%E0%AF%87%E0%AE%B2%E0%AE%BE%E0%AE%A3%E0%AF%8D%E0%AE%AE%E0%AF%88%2B%E0%AE%95%E0%AF%81%E0%AE%B1%E0%AE%BF%E0%AE%A4%E0%AF%8D%E0%AE%A4%E0%AF%81%2B%E0%AE%A8%E0%AF%80%E0%AE%99%E0%AF%8D%E0%AE%95%E0%AE%B3%E0%AF%8D%2B%E0%AE%85%E0%AE%B1%E0%AE%BF%E0%AE%B5%E0%AF%80%E0%AE%B0%E0%AF%8D%E0%AE%95%E0%AE%B3%E0%AE%BE%3F%2B%23%E0%AE%A4%E0%AE%AE%E0%AE%BF%E0%AE%B4%E0%AE%B0%E0%AF%8D_%E0%AE%AA%E0%AF%86%E0%AE%B0%E0%AF%81%E0%AE%AE%E0%AF%88%2B-%2BBBC%2BNews%2B%E0%AE%A4%E0%AE%AE%E0%AE%BF%E0%AE%B4%E0%AF%8D]&x11=[2020-08-19T02%3A19%3A36.000Z]&x12=[2020-08-19T12%3A37%3A43.000Z]&x13=[Environment~Water~Conservation~Tamil%2Blanguage~Tamil%2BNadu]&x14=[0f37fb35-7f9e-4e49-b189-9d7f1d6fb11f~b214dc4c-d30e-4ef7-9da3-33858396aa2b~bce059a1-1a05-4b8d-aac8-381699aaa2f0~eb2f3339-023e-4fca-9b72-c6d77b53452c~fa46a306-d93a-43af-a2b8-b386773e2016]&x16=[WS%20-%20Give%20me%20perspective]&x17=[News]&xto=CS3-33-%5Bwstamil~C~A38B38C42D38E38F39G38wsmktgProudToBeTamil~Tamil_8MS_ProudToBeTamil_Facebook_Traffic_India_E38F39G38%5D-%5BFacebook%5D-%5B23845932568300152%5D-%5B23845937811820152%5D
</details>

image

**Code changes:**
- Encode all ati parameters as part of building the URL: https://github.com/bbc/simorgh/compare/marketing-issue#diff-1c3985090b52f7b568ce9f3651bd02abR201-R207
  - This is the only place encoding takes place as part of building the url, other inconsistent encoding and decoding has been removed in this PR ([href](https://github.com/bbc/simorgh/compare/marketing-issue#diff-1c3985090b52f7b568ce9f3651bd02abR120), [referrer](https://github.com/bbc/simorgh/compare/marketing-issue#diff-1c3985090b52f7b568ce9f3651bd02abR139)) 
- Updated automated tests to use [`toMatchInlineSnaphot`](https://jestjs.io/docs/en/expect#tomatchinlinesnapshotpropertymatchers-inlinesnapshot)
  - This made it easier to update the tests to reflect the new more widespread use of encoding
  - It makes the tests more maintainable in the future, the tests previously were hard coded values that we assembled by hand, `toMatchInlineSnaphot` automates this process.
- Updated integration tests snapshots for ati to reflect the encoded parameters now sent to ATI

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

I think given the sweeping nature of this change if at all possible it would be good to perform a manual regression test for ati, checking the ati analytics solution receives the expected values as marketing would use it. As part of development, I checked the ati tag inspector as above but I don't know whether that is 100% representative of how the ATI solution processes the hit url that is sent.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -5,7 +5,16 @@ SIMORGH_PUBLIC_STATIC_ASSETS_PATH=/
 SIMORGH_ATI_BASE_URL=https://logws1363.ati-host.net?
 SIMORGH_CONFIG_URL=https://config.test.api.bbci.co.uk/
 SIMORGH_INCLUDES_BASE_URL=https://www.test.bbc.com/ws/includes
+SIMORGH_INCLUDES_BASE_AMP_URL=https://news.test.files.bbci.co.uk
```
Comment (karinathomasbbc) [inline .env]: Oops (think this file is in the .gitignore list?)

File: data/tamil/cpsAssets/india-53826135.json [hunk 2]
```diff
@@ -0,0 +1 @@
...ction/_114014287_69397b9e-64fc-418a-9669-7b2d16bf8647.jpg","id":"114014287","path":"/cpsprodpb/131A1/production/_114014287_69397b9e-64fc-418a-9669-7b2d16bf8647.jpg","subType":"index","type":"image","width":976},"language":"ta","locators":{"assetId":"53826135","assetUri":"/tamil/india-53826135","cpsUrn":"urn:bbc:content:assetUri:tamil/india-53826135","curie":"http://www.bbc.co.uk/asset/cebacc47-6b26-413e-9ebf-196bf4fb0b67"},"passport":{"campaigns":[{"campaignId":"5a988e3739461b000e9dabfa","campaignName":"WS - Give me perspective"}],"category":{"categoryId":"http://www.bbc.co.uk/ontologies/applicationlogic-news/News","categoryName":"News"},"taggings":[]},"summary":"நிலம் தீ நீர்வளி விசும்போடு ஐந்தும் கலந்த மயக்கம் உலகம் என்கிறது தொல்காப்பியம். 'நீரின்றி அமையாது உலகு', நீர் 'மிகுனும் குறையினும் நோய் செய்யும்' என்கிறார் வள்ளுவர். நீரில்லாமல் இவ்வுலகம் இல்லை என்பதை பண்டைய தமிழர்கள் அறிந்து வைத்திருந்திருக்கிறார்கள் என்பதற்கான வெளிப்படையான சான்று இது.","timestamp":1597840663000,"type":"cps"},"relatedContent":{"groups":[],"section":{"name":"இந்தியா","subType":"index","type":"simple","uri":"/tamil/india"},"site":{"name":"தமிழில் செய்திகள்","subType":"site","type":"simple","uri":"/tamil"}}}
```
Comment (karinathomasbbc) [inline data/tamil/cpsAssets/india-53826135.json]: Would it be possible to pretty print / reformat this, for easier readability

**Review by karinathomasbbc — APPROVED** [review]
Looks good 👍

File: .env (hunk 1 above)
Comment (andrewscfc) [reply .env]: I don't know what this file is for and why it isn't gitignored; I've reverted the change? I keep meaning to bring this up, we only use the env configs here to my knowledge: https://github.com/bbc/simorgh/tree/latest/envConfig

Comment (karinathomasbbc) [reply .env]: Not 100% sure, I think it's to standardise the start up scripts for all environments - the relevant file is always copied to .env on build? But for some reason the .env file lives in version control 🤷‍♀️ (it's supposed to look like this https://github.com/bbc/simorgh/commit/3fb7b2077777229cfb722c97ee6fdcf7ef730bf6#diff-f579cccc964135c7d644c7b2d3b0d3ec)

**Review by simonsinclair — APPROVED** [review]

**Pull request merged**